apps/api/src/app/events/usecases/trigger-event/trigger-event.usecase.ts

```diff
@@ -277,23 +282,24 @@ export class TriggerEvent {
       content,
     });
 
-    if (
-      command.payload.$phone &&
-      application.channels?.sms?.twillio?.authToken &&
-      application.channels?.sms?.twillio?.accountSid
-    ) {
+    const integration = await this.integrationRepository.findOne({
+      _applicationId: command.applicationId,
+      channel: ChannelTypeEnum.SMS,
+      active: true,
+    });
+
+    if (command.payload.$phone && integration) {
       try {
-        const smsService = new SmsService(
-          application.channels?.sms?.twillio?.authToken,
-          application.channels?.sms?.twillio?.accountSid
-        );
-        const smsResponse = await smsService.sendMessage(
-          command.payload.$phone,
-          application.channels?.sms?.twillio?.phoneNumber,
-          content
-        );
+        const smsHandler = this.smsFactory.getHandler(integration, command.payload.$fromPhone);
+
+        await smsHandler.send({
+          to: command.payload.$phone,
+          from: command.payload.$fromPhone,
```

Comment:
Quick question regarding the new extraction of the origin phone.
In the current case, we have a hardcoded Twilio SMS coupled into application collection.

Do we want to leave it that way?

I can offer two possible solutions: 
1. get it from the payload (in case we don't have any users that will be easy change - that the solutions I added on the draft PR)
2. set it in the integration store client and store it under credentioals.

Reply:
Option 2: We should store it in the integration credentials, since each provider will have a different phone number.

Reply:
I would remove the $fromPhone for now until we finalize the $payload interface to include more data.